types/frisby/index.d.ts

```diff
@@ -1,75 +1,234 @@
-// Type definitions for Frisby v0.8.5
+// Type definitions for Frisby 2.0
 // Project: https://github.com/vlucas/frisby
-// Definitions by: Johnny Li <https://github.com/johnny4753>
+// Definitions by: Christopher E. Woodland <https://github.com/cwoodland>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.2
+// TypeScript Version: 2.6
 
-///<reference types="jasmine"/>
+/// <reference types='jasmine'/>
 
-declare interface Frisby {
-    globalSetup(opts: any): any;
-    timeout(seconds?: number): Frisby;
-    reset(): Frisby;
-    not(): Frisby;
-    addHeader(key: string, content: string): Frisby;
-    addHeaders(headers: object): Frisby;
-    removeHeader(key: string): Frisby;
-    responseType(type: any): Frisby;
-    // HTTP Basic Auth
-    auth(username: string, password: string, digest: boolean): Frisby;
-    // HTTP Request
-    get(uri: string, params?: any): Frisby;
-    patch(uri: string, data?: any, params?: any): Frisby;
-    post(uri: string, data?: any, params?: any): Frisby;
-    put(uri: string, data?: any, params?: any): Frisby;
-    delete(uri: string, data?: any, params?: any): Frisby;
-    head(uri: string, params?: any): Frisby;
-    options(uri: string, params?: any): Frisby;
+// #region Imports
+import { Response } from 'node-fetch'; // Import all definitions from node-fetch.
+//#endregion
 
-    expectMaxResponseTime(milliseconds: number): Frisby;
-    expectStatus(statusCode: number): Frisby;
-    expectHeader(header: string, content: string): Frisby;
-    expectHeaderContains(header: string, content: string): Frisby;
-    expectHeaderToMatch(header: string, pattern: RegExp): Frisby;
-    expectBodyContains(content: string): Frisby;
-    expectJSONTypes(json: any): Frisby;
-    expectJSONTypes(path: string, json: any): Frisby;
-    expectJSON(json: any): Frisby;
-    expectJSON(path: string, json: any): Frisby;
-    expectJSONSchema(path: string, jsonSchema: any): Frisby;
-    expectJSONLength(length: number): Frisby;
-    expectJSONLength(path: string, length: number): Frisby;
-    inspectRequest(): Frisby;
-    inspectResponse(): Frisby;
-    inspectHeaders(): Frisby;
-    inspectBody(): Frisby;
-    inspectJSON(): Frisby;
-    inspectStatus(): Frisby;
-    retry(count: number, delayInMilliseconds?: number): Frisby;
-    waits(delayInMilliseconds: number): Frisby;
-    after(callback: (...args: any[]) => void): Frisby;
-    afterJSON(callback: (...args: any[]) => void): Frisby;
-    exceptionHandler(fn: (...args: any[]) => void): Frisby;
-    toss(): void;
+// #region Joi Methods
+// Reference file: https://github.com/hapijs/joi
+export namespace Joi {
```

Comment:
Rather than rewriting the API surface of Joi, you can just use `export import Joi = require("joi");` and actually match the Joi declaration file.

Reply:
nice!!  I tired export = require("joi") before, but just didn’t know how to do this.
Thank you!

Change made.